samza-core/src/main/java/org/apache/samza/system/inmemory/InMemoryManager.java

```diff
@@ -88,7 +88,7 @@ void put(SystemStreamPartition ssp, Object key, Object message) {
   void put(SystemStreamPartition ssp, IncomingMessageEnvelope envelope) {
     List<IncomingMessageEnvelope> messages = bufferedMessages.get(ssp);
     String offset = String.valueOf(messages.size());
-    if (envelope.getOffset().equals(offset)) {
+    if (!envelope.getOffset().equals(offset)) {
```

Comment:
Would be great to check if the envelope.getOffset() is not null.

Reply:
Done